llmfoundry/models/mpt/modeling_mpt.py

```diff
@@ -1015,9 +1017,18 @@ def forward(self, batch: MutableMapping) -> CausalLMOutputWithPast:
 
     def loss(self, outputs: CausalLMOutputWithPast,
              batch: Mapping) -> torch.Tensor:
-        targets = self.get_targets(batch)
-        return self.loss_fn(outputs.logits.view(-1, outputs.logits.size(-1)),
-                            targets.view(-1))
+        targets = self.get_targets(batch).view(-1)
+        losses = self.loss_fn(outputs.logits.view(-1, outputs.logits.size(-1)),
+                              targets)
+
+        if torch.all(targets == -100):
+            raise ValueError(
+                'All targets are -100. This might be due to all tokens in a sequence being padding tokens.'
+            )
```

Comment:
```suggestion
            loss = losses.sum()
```

Should we return 0 in this case?
maybe log a warning, but Error doesn't make sense.

Reply:
This was done to preserve functionality. The code before this change would result in a NaN being returned as loss, due to dividing by 0. A better way to capture that would be raise an error. That is why I raise this error.